dbux-code/src/treeView/treeData.js

```diff
@@ -0,0 +1,168 @@
+/* eslint-disable class-methods-use-this */
+'use strict'
+
+import vscode, { CommentThreadCollapsibleState } from 'vscode';
+import path from 'path';
+import ExecutionContextType from 'dbux-common/src/core/constants/ExecutionContextType';
+
+const log = (...args) => console.log('[dbux-code][treeData]', ...args)
+
+export class EventNodeProvider {
+
+    constructor(contextData, dataProvider) {
+        this.contextData = contextData;
+        this.dataProvider = dataProvider;
+        this.treeData = this.parseData(this.contextData)
+        this._onDidChangeTreeData = new vscode.EventEmitter();
+        this.onDidChangeTreeData = this._onDidChangeTreeData.event;
+    }
+
+    parseData = (contextData) => {
+
+        log('try parsing contextData', contextData)
+        log('executionContext =', this.dataProvider.collections.executionContexts.getAll())
+        
+        let newRootEvent = new Event('root', new NodePosition(), '', -1);
+        let lastNode = newRootEvent;
+
+        for (let index = 1; index < contextData.length; index++){
+            
+            let element = contextData[index]
+            log('element =', element)
+            let { contextType, stackDepth, contextId, staticContextId } = element;
+            
+            let staticContext = this.dataProvider.collections.staticContexts.getById(staticContextId);
+            log('staticContext =', staticContext)
+            let { programId, name, displayName, loc } = staticContext
+            
+            let programContext = this.dataProvider.collections.staticProgramContexts.getById(programId);
+            log('programContext =', programContext)
+            let { filePath, fileName } = programContext
+
+            let typeName = ExecutionContextType.nameFrom(contextType);
+
+            let newNode = new Event(
+                `${displayName} [${typeName}]`,
+                new NodePosition(filePath, loc.start.line, loc.start.column),
+                fileName,
+                stackDepth
+            );
+
+            if (stackDepth > lastNode.depth){
+                newNode.parent = lastNode;
+            }
+            else if (stackDepth === lastNode.depth){
+                newNode.parent = lastNode.parent;
+            }
+            else if (stackDepth < lastNode.depth){
+                newNode.parent = lastNode.parent.parent;
+            }
```

Comment:
* maintain array `nodesByContext` of all `ContextNode` (where each index is the node's `contextId`)
* get `parentNode` from `nodesByContext[context.parentContextId]`